### PR TITLE
Fix: space checking between tokens (fixes #2211)

### LINKED ISSUE #2211: no-spaced-func counts comments as spaces
I was not expecting [padded-blocks](http://eslint.org/docs/rules/padded-blocks) to result in errors when a comment is the first or last line of a block. Assuming this is a bug but it could also be intentional.

Rule:

```
"padded-blocks": [2, "never"],
```

Code:

```
function fn(a, b) {
  /**
   * comment
   */
  return function () {
    return a + b;
  };
}
```

Error:

```
error  Block must not be padded by blank lines  padded-blocks
```

### PR DESCRIPTION
**Pull request by nzakas**
Not sure about the function name, but `isTokenSpaced()` wasn't quite right, so I went with `isSpaceBetweenTokens()`. Alternative suggestions welcome.

### CODE REVIEW

**Comment by nzakas** [comment]
The Travis failure is the Node 0.12 VM failing to initialize properly. Tests are passed on all other environments.

**Comment by ilyavolodin** [comment]
Force-restarted 0.12 build

File: lib/util/source-code.js [hunk 1]
```diff
@@ -259,8 +253,21 @@ SourceCode.prototype = {
         });
 
         return result;
-    }
+    },
 
+    /**
+     * Determines if two tokens two tokens have at least one whitespace character
```
Comment (lo1tuma) [inline lib/util/source-code.js]: > two tokens ~~two tokens~~

Comment (nzakas) [inline lib/util/source-code.js]: Oops. Copy paste copy paste error.

**Comment by mysticatea** [comment]
Wow, this logic is cool!

File: lib/util/source-code.js (hunk 1 above)
Comment (lo1tuma) [inline lib/util/source-code.js]: :laughing:

**Comment by lo1tuma** [comment]
LGTM

**Pull request merged**